gcp/rakefiles/sh_filter.rb

```diff
@@ -14,4 +16,9 @@ def sh_filter(*cmd)
     end
     puts out
   end
+  exit_status = Process.waitpid2(process.pid)[1].exitstatus
+  if exit_status != 0 and exit_on_nonzero_status
```

Comment:
What is `exit_on_nonzero_status` for? Doesn't look like it has any callers.

Reply:
In case we want to run command with `sh_filter` and get back to rake task execution flow even if command finished with non-zero status. Am I overthinking things again?

Reply:
In general, I would avoid implementing a feature without a specific use case. A long answer to this question: [https://www.martinfowler.com/bliki/Yagni.html](YAGNI).

I think your scenario is plausible. If you have a specific example of wanting to run a command and ignore its return value (e.g. "I'm definitely going to need this for my Locust work"), then this is great. If you don't, I would remove the extra code. However it's not a big deal in this particular case, so it's up to you.